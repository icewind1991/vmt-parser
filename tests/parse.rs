use miette::{GraphicalReportHandler, GraphicalTheme};
use std::fs::read_to_string;
use test_case::test_case;
use vmt_parser::from_str;

#[test_case("tests/data/concretefloor003.vmt")]
#[test_case("tests/data/mvm_backpack.vmt")]
#[test_case("tests/data/water_murky.vmt")]
#[test_case("tests/data/blendrocktograss002.vmt")]
fn test_serde(path: &str) {
    let raw = read_to_string(path).unwrap();
    match from_str(&raw) {
        Ok(result) => insta::assert_ron_snapshot!(path, result),
        Err(e) => {
            let handler = GraphicalReportHandler::new_themed(GraphicalTheme::unicode_nocolor());
            let mut out = String::new();
            handler.render_report(&mut out, &e).unwrap();
            insta::assert_snapshot!(path, out)
        }
    }
}
