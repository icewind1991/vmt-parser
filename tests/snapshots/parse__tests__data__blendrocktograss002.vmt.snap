---
source: tests/parse.rs
expression: result
---
worldvertextransition(WorldVertexTransitionMaterial(
  r#$basetexture: "cp_mountainlab/nature/rockwall001",
  r#$basetexture2: "cp_mountainlab/nature/grass001",
  r#$blendmodulatetexture: Some("nature/grass_blendmask"),
  r#$decal: false,
  r#$detail: None,
  r#$surfaceprop: Some("dirt"),
  r#$basetexturetransform: "center 0.5 0.5 scale 1 1 rotate 0 translate 0 0",
  r#$color: Vec3((1.0, 1.0, 1.0)),
  r#$decalscale: 1.0,
  r#$decalscale: Vec2((4.0, 4.0)),
  r#$detailblendfactor: 1.0,
  r#$detailblendmode: 1,
  r#$modelmaterial: None,
  r#$pointsamplemagfilter: false,
  r#$seamless_scale: 1.0,
  r#$alpha: 1.0,
  r#$alphatest: false,
  r#$distancealpha: false,
  r#$nocull: false,
  r#$translucent: false,
  r#$bumpmap: Some("nature/rockwall009_height-ssbump"),
  r#$bumpmap2: None,
  r#$lightwarptexture: None,
  r#$selfillum: false,
  r#$ssbump: true,
  r#$envmap: None,
  r#$phong: false,
  r#$nofog: false,
  r#$ignorez: false,
))
